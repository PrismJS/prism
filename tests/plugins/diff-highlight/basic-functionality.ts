import { createTestSuite } from '../../helper/prism-dom-util';

const jsCode = `
@@ -4,6 +4,5 @@
-    let foo = bar.baz([1, 2, 3]);
-    foo = foo + 1;
+    const foo = bar.baz([1, 2, 3]) + 1;
     console.log(\`foo: \${foo}\`);
`.trim();

describe('Diff Highlight', () => {
	const { it } = createTestSuite({
		languages: ['javascript'],
		plugins: 'diff-highlight'
	});

	it('should work with language-diff', ({ util }) => {
		util.assert.highlight({
			language: 'diff',
			code: jsCode
		});
	});

	it('should work with language-diff-javascript', ({ util }) => {
		util.assert.highlight({
			language: 'diff-javascript',
			code: jsCode
		});
	});

	it('should work with aliases', ({ util }) => {
		util.assert.highlight({
			language: 'diff-js',
			code: jsCode
		});
	});
});
