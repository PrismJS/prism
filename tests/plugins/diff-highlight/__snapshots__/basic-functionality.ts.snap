// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diff Highlight should work with aliases 1`] = `
"<span class=\\"token coord\\">@@ -4,6 +4,5 @@</span>
<span class=\\"token deleted-sign deleted\\">
	<span class=\\"token prefix deleted\\">-</span>
	<span class=\\"token keyword\\">let</span>
	foo
	<span class=\\"token operator\\">=</span>
	bar
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">baz</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token punctuation\\">[</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">2</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">3</span>
	<span class=\\"token punctuation\\">]</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token punctuation\\">;</span>
	<span class=\\"token prefix deleted\\">-</span>
	foo
	<span class=\\"token operator\\">=</span>
	foo
	<span class=\\"token operator\\">+</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">;</span>
</span>
<span class=\\"token inserted-sign inserted\\">
	<span class=\\"token prefix inserted\\">+</span>
	<span class=\\"token keyword\\">const</span>
	foo
	<span class=\\"token operator\\">=</span>
	bar
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">baz</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token punctuation\\">[</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">2</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">3</span>
	<span class=\\"token punctuation\\">]</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token operator\\">+</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">;</span>
</span>
<span class=\\"token unchanged\\">
	<span class=\\"token prefix unchanged\\"></span>
	console
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">log</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token template-string\\">
		<span class=\\"token template-punctuation string\\">\`</span>
		<span class=\\"token string\\">foo:</span>
		<span class=\\"token interpolation\\">
			<span class=\\"token interpolation-punctuation punctuation\\">\${</span>
			foo
			<span class=\\"token interpolation-punctuation punctuation\\">}</span>
		</span>
		<span class=\\"token template-punctuation string\\">\`</span>
	</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token punctuation\\">;</span>
</span>
"
`;

exports[`Diff Highlight should work with language-diff 1`] = `
"<span class=\\"token coord\\">@@ -4,6 +4,5 @@</span>
<span class=\\"token deleted-sign deleted\\">
	<span class=\\"token prefix deleted\\">-</span>
	let foo = bar.baz([1, 2, 3]);
	<span class=\\"token prefix deleted\\">-</span>
	foo = foo + 1;
</span>
<span class=\\"token inserted-sign inserted\\">
	<span class=\\"token prefix inserted\\">+</span>
	const foo = bar.baz([1, 2, 3]) + 1;
</span>
<span class=\\"token unchanged\\">
	<span class=\\"token prefix unchanged\\"></span>
	console.log(\`foo: \${foo}\`);
</span>
"
`;

exports[`Diff Highlight should work with language-diff-javascript 1`] = `
"<span class=\\"token coord\\">@@ -4,6 +4,5 @@</span>
<span class=\\"token deleted-sign deleted\\">
	<span class=\\"token prefix deleted\\">-</span>
	<span class=\\"token keyword\\">let</span>
	foo
	<span class=\\"token operator\\">=</span>
	bar
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">baz</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token punctuation\\">[</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">2</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">3</span>
	<span class=\\"token punctuation\\">]</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token punctuation\\">;</span>
	<span class=\\"token prefix deleted\\">-</span>
	foo
	<span class=\\"token operator\\">=</span>
	foo
	<span class=\\"token operator\\">+</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">;</span>
</span>
<span class=\\"token inserted-sign inserted\\">
	<span class=\\"token prefix inserted\\">+</span>
	<span class=\\"token keyword\\">const</span>
	foo
	<span class=\\"token operator\\">=</span>
	bar
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">baz</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token punctuation\\">[</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">2</span>
	<span class=\\"token punctuation\\">,</span>
	<span class=\\"token number\\">3</span>
	<span class=\\"token punctuation\\">]</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token operator\\">+</span>
	<span class=\\"token number\\">1</span>
	<span class=\\"token punctuation\\">;</span>
</span>
<span class=\\"token unchanged\\">
	<span class=\\"token prefix unchanged\\"></span>
	console
	<span class=\\"token punctuation\\">.</span>
	<span class=\\"token function\\">log</span>
	<span class=\\"token punctuation\\">(</span>
	<span class=\\"token template-string\\">
		<span class=\\"token template-punctuation string\\">\`</span>
		<span class=\\"token string\\">foo:</span>
		<span class=\\"token interpolation\\">
			<span class=\\"token interpolation-punctuation punctuation\\">\${</span>
			foo
			<span class=\\"token interpolation-punctuation punctuation\\">}</span>
		</span>
		<span class=\\"token template-punctuation string\\">\`</span>
	</span>
	<span class=\\"token punctuation\\">)</span>
	<span class=\\"token punctuation\\">;</span>
</span>
"
`;
